cmd/kubectl-gadget/filetop.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/filetop/types"
+)
+
+var (
+	nodeEvents map[string][]types.Event
+	mu         sync.Mutex
+	maxrows    int
+	noclear    bool
+	sortby_str string
+	sortby     types.SortBy
+)
+
+var filetopCmd = &cobra.Command{
+	Use:   "filetop",
+	Short: "Trace reads and writes by file, with container details",
+	Run: func(cmd *cobra.Command, args []string) {
+		nodeEvents = make(map[string][]types.Event)
+
+		config := &utils.TraceConfig{
+			GadgetName:       "filetop",
+			Operation:        "start",
+			TraceOutputMode:  "Stream",
+			TraceOutputState: "Started",
+			CommonFlags:      &params,
+		}
+
+		startprint()
+
+		err := utils.RunTraceStreamCallback(config, callback)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s\n", err)
+			os.Exit(1)
+		}
+	},
+	SilenceUsage: true,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		sortby, err = types.ParseSortBy(sortby_str)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	},
+}
+
+func init() {
+	filetopCmd.Flags().BoolVarP(&noclear, "noclear", "C", false, "don't clear the screen")
```

Comment:
I wonder if that should be automatic by detecting if stdout is a terminal.

https://pkg.go.dev/golang.org/x/term#IsTerminal

```
if term.IsTerminal(int(os.Stdout.Fd()))) {
  utils.ClearScreen()
} else {
  fmt.Println("")
}
```

Reply:
Yes, makes sense. I removed the flag and used this approach instead.